### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ikhoon**
- Brave 5.15.0 -> 5.16.0
- Brotli4J 1.11.0 -> 1.12.0
- Curator 5.4.0 -> 5.5.0
- Dropwizard 2.1.5 -> 2.1.6
- Dropwizard Metrics 4.2.18 -> 4.2.19
- GraphQL Java 20.0 -> 20.4
- GraphQL Kotlin 6.4.0 -> 6.5.2
- gRPC-Java 1.54.0 -> 1.56.0
- Jackson 2.14.2 -> 2.15.2
- java-jwt 4.3.0 -> 4.4.0
- Kafka client 3.4.0 -> 3.4.1
- Kotlin 1.8.10 -> 1.8.22
- Kotlin Coroutine 1.6.4 -> 1.7.1
- Micrometer 1.10.5 -> 1.11.1
- Netty 4.1.92 -> 4.1.93
  - io_uring 0.0.20 -> 0.0.21
- protobuf-jackson 2.0.0 -> 2.2.0
- Reactive gRPC 1.2.3 -> 1.2.4
- Reactor 3.5.1 -> 3.5.7
- Sangria 3.5.3 -> 4.0.0
- Sangria slowlog 2.0.5 -> 3.0.0
- Scala 2.12.17 -> 2.12.18, 2.13.10 -> 2.13.11, 3.2.2 -> 3.3.0
- scala-collection-compat 2.9.0 -> 2.10.0
- Spring 6.0.6 -> 6.0.9
- Spring Boot 2.7.10 -> 2.7.12, 3.0.5 -> 3.1.0
- Tomcat 10.1.4 -> 10.1.10
- Build
  - Dagger 2.45 -> 2.46.1
  - Error Prone 2.18.0 -> 2.19.1
  - Error Prone Gradle plugin 3.0.1 -> 3.1.0
  - gax-grpc 2.20.1
  - Groovy 3.0.16 -> 3.0.17
  - Guava 31.1 -> 32.0.1
  - jmh-gradle-plugin 0.7.0 -> 0.7.1
  - json-unit 2.37.0 -> 2.38.0
  - jsoup 1.15.3 -> 1.16.1
  - JUnit 5.9.2 -> 5.9.3
  - ktlint-gradle-plugin 11.3.2 -> 11.4.0
  - node-gradle-plugin 3.5.1 -. 5.0.0
  - OkHttp 4.10.0 -> 4.11.0

### CODE REVIEW

**Review by jrhee17 — APPROVED** [review]
Thanks @ikhoon 👍 👍 👍

File: core/src/test/java/com/linecorp/armeria/client/RestClientTest.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ void restApi(RestClient restClient) {
                     preparation = restClient.delete("/rest/{id}");
                     break;
             }
-            assert preparation != null;
+            assertThat(preparation).isNotNull();
```
Comment (trustin) [inline core/src/test/java/com/linecorp/armeria/client/RestClientTest.java:112]: Thanks for cleaning this up, @ikhoon!

**Review by trustin — APPROVED** [review]
Thanks a lot, @ikhoon! :bow:

**Review by minwoox — APPROVED** [review]
Thanks, @ikhoon! 👍 👍 👍

File: core/src/test/java/com/linecorp/armeria/client/RestClientTest.java (hunk 1 above)
Comment (ikhoon) [reply core/src/test/java/com/linecorp/armeria/client/RestClientTest.java:112]: New [Error Prone ](https://errorprone.info/bugpattern/UseCorrectAssertInTests) does not allow `assert` in tests.

**Comment by ikhoon** [comment]
The following dependencies are updated after the final review. [`0571c7a` (#4939)](https://github.com/line/armeria/pull/4939/commits/0571c7a98ae3fa6b929732666de0bc1eddc05216)
- Guava 32.0.0 -> 32.0.1
- gRPC-Java 1.55.1 -> 1.56.0
- Micrometer 1.11.0 -> 1.11.1
- Reactive gRPC 1.2.3 -> 1.2.4
- Reactor 3.5.6 -> 3.5.7
- Tomcat 10.1.9 -> 10.1.10

**Pull request merged**